crates/uv-configuration/src/trusted_host.rs

```diff
@@ -0,0 +1,95 @@
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+/// A trusted host, which could be a host or a host-port pair.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum TrustedHost {
+    Host(String),
+    HostPort(String, u16),
+}
```

Comment:
I think you can use `Realm` in place of this (`crates/uv-auth/src/realm.rs`) — I think you can just move the `from_str` implementation over.

Reply:
"should" because it's our canonical way to do this and there's `From<&Url> for Realm`.

Reply:
Doesn't that require a scheme? These values don't require a scheme.

Reply:
It does yeah, hm. But if someone provides a scheme here (which is allowed per the parser), shouldn't that match be enforced?

Reply:
The scheme could become optional on `Realm`, I'm not sure if that causes problems. I don't feel strongly if it's complicated. There's not a lot of complexity in the matching anymore (`Realm` got simplified at some point, relying on some behaviors from `reqwest`)

Reply:
Yeah I agree. Should we assume HTTPS if omitted...? It would be strange to put that in `FromStr`, but it could make sense for _this_ setting. Or could scheme be optional on `Realm`?

Reply:
I'm not sure, I feel like if you do `--trusted-host example.com` and we make a request to `http://example.com`— does skipping SSL verification even matter or make sense? It seems reasonable to assume `https` since it's only relevant there?

Reply:
And I guess another point this raises: should we allow `--trusted-host http://example.com`? What does that mean?

Reply:
I suppose we could reject `http`.

Reply:
We should allow `http://`. It's normal practice to run internal pypi mirror that runs on http. That's the whole point of `trusting` a host.

Reply:
Same applies to, lets say the mirror is https but uses a custom/internal/enterprise CA.

You would want to install from it without verifying the cert, for example inside a Docker build, where you dont want to be adding custom CA's in the container.

Reply:
If you're running over `http://`, why would certificate validity matter?

Reply:
The following use-cases should work with --trusted-host/--insecure-host or whichever is the final name.
- http://example.com
- https://example.com
- https://example.com (Don't verify CA)

Reply:
Sorry, I'm not following. Why is `--trusted-host` necessary _at all_ for an `http://` domain?

Reply:
I guess pip warns if you try to use `http://`? We don't do that. I don't think you need `--trusted-host` at all for `http://` domains with uv. This is only relevant to `https://`.

Reply:
You are right, there's no verification needed.

Reply:
Yeah I feel like setting an index URL to `http://example.com` works today (e.g., I'm pretty sure it was fine in #5726) but setting `ALL_PROXY` didn't work.

Reply:
We might need to test what happens if you set `HTTPS_PROXY` or `ALL_PROXY` to an `http://` URL?

Reply:
@zanieb That's typically what I use when using something like Squid-Proxy + Pypi. I set both `HTTP_PROXY` and `HTTPS_PROXY` to "http://squid-hostname`, and let Squid handle the request/redirect/etc.

Reply:
Will do.

Reply:
> We should allow `http://`. It's normal practice to run internal pypi mirror that runs on http. That's the whole point of `trusting` a host.

This is a reason as to why I'm suggesting a rename away from `--trusted-host`  as it's not about trusting a host, rather skipping TLS verification 😅

Reply:
@zanieb - I opted for now to keep `TrustedHost`, because the requirements are different than `Realm`... `Realm` requires a scheme, `TrustedHost` requires a host (`Realm` does not), etc. I was also hesitant to make `uv-auth` a dependency everywhere, and add `Schemars` as a dep to that crate.